pkg/scheduler/framework/util.go

```diff
@@ -262,3 +264,35 @@ func (nl *NodeLister) List() ([]*v1.Node, error) {
 	}
 	return nodes, nil
 }
+
+func nodeIsReady(obj *v1.Node) bool {
+	conditionMap := make(map[v1.NodeConditionType]*v1.NodeCondition)
+	NodeAllConditions := []v1.NodeConditionType{v1.NodeReady}
+	for i := range obj.Status.Conditions {
+		cond := obj.Status.Conditions[i]
+		conditionMap[cond.Type] = &cond
+	}
+
+	var status []string
+	for _, validCondition := range NodeAllConditions {
+		if condition, ok := conditionMap[validCondition]; ok {
+			if condition.Status == v1.ConditionTrue {
+				status = append(status, string(condition.Type))
+			} else {
+				status = append(status, "Not"+string(condition.Type))
+			}
+		}
+	}
+	if len(status) == 0 {
+		status = append(status, "Unknown")
+	}
+	if obj.Spec.Unschedulable {
+		status = append(status, "SchedulingDisabled")
+	}
+
+	if len(status) != 1 || (len(status) == 1 && !slices.Contains(status, string(v1.NodeReady))) {
```

Comment:
If the `NodeReady` condition is false and the status length is 1, then `nodeIsReady` will also return true, is it what we expect? We should not use `slices.Contains` to judge it

Reply:
Based on everyone's opinions, I have simplified it. Thank you